pkg/runtime/wfengine/workflow.go

```diff
@@ -133,7 +142,7 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 	if exists {
 		runtimeState := getRuntimeState(actorID, state)
 		if runtimeState.IsCompleted() {
-			state = &workflowState{}
+			state.Reset()
```

Comment:
Please cover reset case.

Reply:
Done. I added the `TestRecreateCompletedWorkflow` to cover this, and also fixed a bug that this new test exposed.